src/librustc_resolve/diagnostics.rs

```diff
@@ -1445,29 +1446,35 @@ fn find_span_immediately_after_crate_name(
 crate fn show_candidates(
     err: &mut DiagnosticBuilder<'_>,
     // This is `None` if all placement locations are inside expansions
-    span: Option<Span>,
+    use_placement_span: Option<Span>,
     candidates: &[ImportSuggestion],
     better: bool,
     found_use: bool,
 ) {
     if candidates.is_empty() {
         return;
     }
+
+    let (determiner, kind) = if candidates.len() == 1 {
+        ("this", candidates[0].descr)
+    } else {
+        ("one of these", "items")
+    };
+
     // we want consistent results across executions, but candidates are produced
     // by iterating through a hash map, so make sure they are ordered:
     let mut path_strings: Vec<_> =
         candidates.iter().map(|c| path_names_to_string(&c.path)).collect();
     path_strings.sort();
     path_strings.dedup();
 
-    let better = if better { "better " } else { "" };
-    let msg_diff = match path_strings.len() {
-        1 => " is found in another module, you can import it",
-        _ => "s are found in other modules, you can import them",
+    let msg = if better {
+        format!("consider using {} {} instead", determiner, kind)
```

Comment:
```suggestion
        format!("consider importing {} {}{}", determiner, kind, instead)
```
I'd say "importing" her too because we are recommending to import things in both cases.